source/common/logger/ur_logger.hpp

```diff
@@ -107,7 +112,8 @@ template <typename T> inline std::string toHex(T t) {
 ///                            to be printed immediately as they occur
 ///             - output: stderr
 inline Logger create_logger(std::string logger_name, bool skip_prefix,
-                            bool skip_linebreak) {
+                            bool skip_linebreak,
+                            std::string default_log_level) {
```

Comment:
It might be simpler to take `logger::Level default_level = logger::Level::QUIET` as a parameter instead of changing how the env variables are parsed.

Reply:
It might be, yeah. I was originally hoping that being able to more generically provide default values for environment variables would be useful across more of the project. But so far it's just coming down to the default logging configuration (well, level) and the device selector.

I don't have a strong preference either way.

Reply:
Up to you. I think I'd slightly prefer passing variables instead of default env string. Less error prone and simpler.

Reply:
Yep, agreed. I've pushed that change now. I had to do a bit of plumbing to allow users to set `quiet` (i.e., back to the default) via the environment variable, as it isn't currently a valid value.